Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,802 @@
+                        RequiereCharacterSelected = bool.Parse(reader.GetAttribute("RequiereCharacterSelected")),
+                        DevModeOnly = bool.Parse(reader.GetAttribute("DevModeOnly") ?? "false")
+                    });
+                    break;
+                case "GetSpriteName":
+                    getSpriteNameAction = reader.GetAttribute("FunctionName");
+                    break;
+
+                case "JobSpotOffset":
+                    JobSpotOffset = new Vector2(
+                    int.Parse(reader.GetAttribute("X")),
+                    int.Parse(reader.GetAttribute("Y")));
+                    break;
+                case "Params":
+                    ReadXmlParams(reader);  // Read in the Param tag
+                    break;
+
+                case "LocalizationCode":
+                    reader.Read();
+                    LocalizationCode = reader.ReadContentAsString();
+                    break;
+
+                case "UnlocalizedDescription":
+                    reader.Read();
+                    UnlocalizedDescription = reader.ReadContentAsString();
+                    break;
+            }
+        }
+    }
+    
+    /// <summary>
+    /// Reads the specified XMLReader (pass it to <see cref="ReadXmlParams(XmlReader)"/>)
+    /// This is used to load utility from a save file.
+    /// </summary>
+    /// <param name="reader">The XML reader to read from.</param>
+    public void ReadXml(XmlReader reader)
+    {
+        // X, Y, and objectType have already been set, and we should already
+        // be assigned to a tile.  So just read extra data if we have any.
+        if (!reader.IsEmptyElement)
+        {
+            ReadXmlParams(reader);
+        }
+    }
+
+    /// <summary>
+    /// Reads the XML for parameters that this utility has and assign it to the utility.
+    /// </summary>
+    /// <param name="reader">The reader to read the parameters from.</param>
+    public void ReadXmlParams(XmlReader reader)
+    {
+        // X, Y, and objectType have already been set, and we should already
+        // be assigned to a tile.  So just read extra data.
+        utilParameters = Parameter.ReadXml(reader);
+    }
+
+    /// <summary>
+    /// Gets the utility's Parameter structure from a string key.
+    /// </summary>
+    /// <returns>The Parameter value..</returns>
+    public Parameter GetParameters()
+    {
+        return utilParameters;
+    }
+
+    /// <summary>
+    /// How many jobs are linked to this utility.
+    /// </summary>
+    /// <returns>The number of jobs linked to this utility.</returns>
+    public int JobCount()
+    {
+        return jobs.Count;
+    }
+
+    /// <summary>
+    /// Link a job to the current utility.
+    /// </summary>
+    /// <param name="job">The job that you want to link to the utility.</param>
+    public void AddJob(Job job)
+    {
+        // FIXME: Do we even need this?
+//        job.utility = this;
```

Comment:
we could make job.furniture and job.utility something like job.buildable so that it works for both. You might need to know about it for repair jobs.